tests/testing_config/custom_components/test/weather.py

```diff
@@ -298,6 +298,37 @@ async def async_forecast_hourly(self) -> list[Forecast] | None:
         ]
 
 
+class MockWeatherMockLegacyForecastOnly(MockWeather):
```

Comment:
This, and other classes used for testing the weather integration, really should not be here.
We can clean it up in a separate PR though.